src/QsCompiler/Optimizations/Utils/HelperFunctions.fs

```diff
@@ -50,19 +51,22 @@ let rec internal isLiteral (callables: ImmutableDictionary<QsQualifiedName, QsCa
 /// Otherwise, returns constants without any changes.
 /// If the given variable is already defined, its name is shadowed in the current scope.
 /// Throws an InvalidOperationException if there aren't any scopes on the stack.
-let internal defineVar check constants (name, value) =
-    if not (check value) then constants else constants |> Map.add name value
+let internal defineVar check (constants : IDictionary<_,_>) (name, value) =
+    //if not (check value) then constants else constants |> Map.add name value
```

Comment:
Should this commented out code still be here?

Reply:
No, thanks!